### PR TITLE
Fixes #7934: Merge inventory from initial promises and system technique

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/7934

### CODE REVIEW

File: initial-promises/node-server/inventory/1.0/fetchFusionTools.cf [hunk 1]
```diff
@@ -38,10 +37,9 @@ bundle agent fetchFusionTools
         classes => rudder_common_classes("fetchFusionTools_install_curl"),
         comment => "Installing curl using yum";
 
-  reports:
+  methods:
     fetchFusionTools_install_curl_error::
-      "@@Inventory@@result_error@@inventory-all@@inventory-all@@00@@tools@@None@@${g.execRun}##${g.uuid}@#Installing 'curl' failed. Inventory registration will fail if curl is unavailable";
-
+      "any" usebundle => rudder_common_report("Inventory", "result_error", "hasPolicyServer-root@@common-root@@00", "inventory", "None", "Installing 'curl' failed. Inventory registration will fail if curl is unavailable");
```
Comment (jooooooon) [inline initial-promises/node-server/inventory/1.0/fetchFusionTools.cf]: The "rule@@directive@@serial" bit doesn't match here

File: techniques/system/inventory/1.0/fusionAgent.st [hunk 2]
```diff
@@ -427,49 +427,79 @@ bundle agent turnUsersToUnicode
...
+      "RUDDERUUID"     string => execresult("/system/xbin/sqlite3 /data/data/com.android.providers.settings/databases/settings.db \"select value from secure where name = 'android_id'\"", "noshell");
+    aix::
+      "RUDDERUUID"     string => execresult("/usr/bin/uname -f", "noshell");
+    solaris::
+      "RUDDERUUID"     string => execresult("smbios | grep UUID | sed 's/  UUID: //'", "useshell");
+    freebsd::
+      "RUDDERUUID"     string => execresult("/usr/local/sbin/dmidecode -s system-uuid", "noshell");
+    linux.!xen::
+      "RUDDERUUID"     string => execresult("/usr/sbin/dmidecode -s system-uuid", "noshell");
+    linux.xen::
```
Comment (jooooooon) [inline techniques/system/inventory/1.0/fusionAgent.st]: This is missing the class xenrudderuid (and a note explaining that this comes from a regex evaluated in the classes: section below)

File: techniques/system/inventory/1.0/fusionAgent.st [hunk 3]
```diff
@@ -604,8 +603,11 @@ bundle agent sendInventory
 {
 
   vars:
-
+&if(INITIAL)&
       "download_endpoint"         string => "${g.inventory_upload_protocol}://${server_info.cfserved}/inventory-updates/";
+&else&
+      "download_endpoint"         string => "${g.inventory_upload_protocol}://${server_info.cfserved}/inventories/";
```
Comment (jooooooon) [inline techniques/system/inventory/1.0/fusionAgent.st]: This is the opposite (inventory-updates is for *not* initial)

File: techniques/system/inventory/1.0/fusionAgent.st [hunk 4]
```diff
@@ -151,7 +151,7 @@ bundle common inventory
   classes:
 
     windows::
-
+      "rudder_inventory_cpuid_tool_present"    expression => fileexists("\"${g.rudder_tools}\cpuid-windows-V1.0.vbs\"");
```
Comment (jooooooon) [inline techniques/system/inventory/1.0/fusionAgent.st]: This is not necessary as it is already done in the "any" case 2 lines below

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

**Pull request merged**